_search-plugins/retrieve-specific-fields.md

```diff
@@ -0,0 +1,877 @@
+        "_id": "1",
+        "_score": 1.0,
+        "_source": {
+          "title": "OpenSearch Basics",
+          "author": "John Doe",
+          "publication_date": "2021-01-01"
+        }
+      }
+    ]
+  }
+}
+```
+
+### Including and excluding fields in the same search
+
+There might be cases where both `include` and `exclude` parameters are necessary. See following examples which demonstrates the usefulness of this option.
+
+If you have an index `products` with the following document:
+```
+{
+  "product_id": "123",
+  "name": "Smartphone",
+  "category": "Electronics",
+  "price": 699.99,
+  "description": "A powerful smartphone with a sleek design.",
+  "reviews": [
+    {
+      "user": "john_doe",
+      "rating": 5,
+      "comment": "Great phone!",
+      "date": "2023-01-01"
+    },
+    {
+      "user": "jane_doe",
+      "rating": 4,
+      "comment": "Good value for money.",
+      "date": "2023-02-15"
+    }
+  ],
+  "supplier": {
+    "name": "TechCorp",
+    "contact_email": "support@techcorp.com",
+    "address": {
+      "street": "123 Tech St",
+      "city": "Techville",
+      "zipcode": "12345"
+    }
+  },
+  "inventory": {
+    "stock": 50,
+    "warehouse_location": "A1"
+  }
+}
+```
+
+If you want to perform a search, but in the response you only want to include the `name`, `price`, `reviews`, and `supplier` fields and exclude any `contact_email` fields from the `supplier` object, and `comment` fields from the `reviews` object. Following is the search you can run:
+
+```
+GET /products/_search
+{
+  "_source": {
+    "includes": ["name", "price", "reviews.*", "supplier.*"],
+    "excludes": ["reviews.comment", "supplier.contact_email"]
+  },
+  "query": {
+    "match": {
+      "category": "Electronics"
+    }
+  }
+}
+```
+
+Expected response:
+
+```
+{
+  "hits": {
+    "hits": [
+      {
+        "_source": {
+          "name": "Smartphone",
+          "price": 699.99,
+          "reviews": [
+            {
+              "user": "john_doe",
+              "rating": 5,
+              "date": "2023-01-01"
+            },
+            {
+              "user": "jane_doe",
+              "rating": 4,
+              "date": "2023-02-15"
+            }
+          ],
+          "supplier": {
+            "name": "TechCorp",
+            "address": {
+              "street": "123 Tech St",
+              "city": "Techville",
+              "zipcode": "12345"
+            }
+          }
+        }
+      }
+    ]
+  }
+}
+```
+
+## Using scripted fields
+
+The `script_fields` parameter in OpenSearch allows you to include custom fields in your search results, where the values of these fields are computed using scripts. This can be useful for calculating values on the fly based on the data in the document. You can also retrieve `derived fields` using similar approach, see [Retrieving fields]({{site.url}}{{site.baseurl}}/field-types/supported-field-types/derived/#retrieving-fields) for further details.
```

Comment:
```suggestion
The `script_fields` parameter allows you to include custom fields whose values are computed using scripts in your search results. This can be useful for calculating values dynamically based on the data in the document. You can also retrieve `derived fields` using a similar approach. For more information, see [Retrieving fields]({{site.url}}{{site.baseurl}}/field-types/supported-field-types/derived/#retrieving-fields).
```